integrated_channels/integrated_channel/learner_data.py

```diff
@@ -134,6 +133,34 @@ def collect_learner_data(self):
                 is_passing=is_passing,
             )
 
+    def _should_skip_enrollment(self, enterprise_enrollment):
+        """
+        Specifies whether to skip data collection of the enterprise enrollment due to any configuration reason.
+
+        :param enterprise_enrollment: The enterprise enrollment whose configuration is to be checked against.
+        :return: True if we should skip, False otherwise.
+        """
+        return any([
+            not enterprise_enrollment.consent_available,
+            self._disabled_audit_track_data_reporting(enterprise_enrollment),
+        ])
+
+    def _disabled_audit_track_data_reporting(self, enterprise_enrollment):
+        """
+        Specifies whether to skip data collection due to audit track data reporting not being enabled.
+
+        :param enterprise_enrollment: The enterprise enrollment whose reporting configuration is to be checked against.
+        :return: True if we should skip, False otherwise.
+        """
+        if not self.enterprise_customer.enables_audit_data_reporting:
+            if self.course_enrollment_api is None:
+                self.course_enrollment_api = EnrollmentApiClient()
+            course_enrollment = self.course_enrollment_api.get_course_enrollment(
+                enterprise_enrollment.enterprise_customer_user.username,
+                enterprise_enrollment.course_id
+            )
+            return course_enrollment['mode'] in ['audit', 'honor'] if course_enrollment is not None else True
```

Comment:
@UmanShahzad I think I understand what you're trying to do here, but it wasn't obvious right away:

* If we're dealing with a `verified` enrollment, data collection should proceed as normal; this method should return `False`.
* If we're dealing with an `audit` or an `honor` enrollment, data collection should:
    * proceed as normal if the Enterprise customer enables audit data reporting; this method should return `False`.
    * be skipped if the Enterprise customer does *not* enable audit data reporting; this method should return `True`.

The current approach is one way of implementing this behavior (although strictly speaking the method returns `None` instead of a Boolean if the Enterprise customer enables audit data reporting), but maybe there's a way to write this that is a bit more accessible/easier to parse on first read? If not, we should at least extend the docstring to explain what's going on; it glosses over some important details at the moment.